robot/rover/internal_comms/include/CommandCenter.h

```diff
@@ -2,39 +2,79 @@
 #define INTERNAL_COMMS_COMMANDCENTER_H
 //
 // Created by cedric on 2020-10-14.
+// And mostly written by Tim
 //
 
 #include "Arduino.h"
+#include <etl/queue.h>
 
 namespace internal_comms
 {
+   typedef struct {
+       uint8_t commandID; // ID of command
+       uint8_t* rawArgs; // Byte array with the bytes of the arguments
+       uint16_t rawArgsLength; // Number of bytes in the rawArgs array
+       bool isValid; // Whether the command is valid
+   } Command;
+
```

Comment:
we could have

typedef struct {
      Message* message;
      bool isValid;
} Command;

To avoid repeated code here

Reply:
would the data be accessible using cmd.*?